azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -13,47 +13,184 @@
 # limitations under the License.
 #
 # Requires Python 2.6+ and Openssl 1.0+
+import threading
+from datetime import datetime as dt
 
-import time
-
-from azurelinuxagent.common import logger
-from azurelinuxagent.common.cgroup import CpuCgroup, MemoryCGroup
 
 class CGroupsTelemetry(object):
     """
     """
-    _tracked = {}
+    _tracked = []
+    _cgroup_metrics = {}
+    _rlock = threading.RLock()
+
+    @staticmethod
+    def _get_metrics_list(metric):
+        return [metric.average(), metric.min(), metric.max(), metric.median(), metric.count(), metric.first_polltime(),
+                metric.last_polltime()]
+
+    @staticmethod
+    def _process_cgroup_metric(cgroup_metrics):
+        current_memory_usage, max_memory_levels, current_cpu_usage = cgroup_metrics.get_metrics()
+        processed_extension = {
+            "memory": {
+                "cur_mem": CGroupsTelemetry._get_metrics_list(current_memory_usage),
+                "max_mem": CGroupsTelemetry._get_metrics_list(max_memory_levels)
```

Comment:
What's the need for average, min, max, median, count... for max memory?

Reply:
First - Implementation-wise it was easier to manage all the metrics as of type Metric. I am thinking of simplifying it with the next round of change, but this makes it simple.

```
class CgroupMetrics(object):
    def __init__(self):
        self.current_memory_usage = Metric()
        self.max_memory_levels = Metric()
        self.current_cpu_usage = Metric()
```

Also, the Max of the Max list can still give the absolute Max, and its a value which can change as well.